src/main/java/witchinggadgets/common/util/handler/EventHandler.java

```diff
@@ -533,4 +540,75 @@ private boolean addToPlayerInventory(EntityPlayer entityPlayer, ItemStack item)
 
         return false;
     }
+
+    @SubscribeEvent
+    public void onPlayerAttacking(AttackEntityEvent event) {
+        for (Object[] gear : buildEventItemList(event.entityPlayer)) {
+            ItemStack stack = (ItemStack) gear[0];
+            itemTriggerEvent(stack, (EntityPlayer) event.entityLiving, event);
+        }
+    }
+
+    @SubscribeEvent
+    public void onPlayerJump(LivingJumpEvent event) {
+        if (event.entityLiving instanceof EntityPlayer)
+            for (Object[] gear : buildEventItemList((EntityPlayer) event.entityLiving)) {
+                ItemStack stack = (ItemStack) gear[0];
+                itemTriggerEvent(stack, (EntityPlayer) event.entityLiving, event);
+            }
+    }
+
+    @SubscribeEvent
+    public void onPlayerFall(LivingFallEvent event) {
+        if (event.entityLiving instanceof EntityPlayer)
+            for (Object[] gear : buildEventItemList((EntityPlayer) event.entityLiving)) {
+                ItemStack stack = (ItemStack) gear[0];
+                itemTriggerEvent(stack, (EntityPlayer) event.entityLiving, event);
+            }
+    }
+
+    @SubscribeEvent
+    public void onPlayerTargeted(LivingSetAttackTargetEvent event) {
+        if (event.target instanceof EntityPlayer)
+            for (Object[] gear : buildEventItemList((EntityPlayer) event.target)) {
+                ItemStack stack = (ItemStack) gear[0];
+                itemTriggerEvent(stack, (EntityPlayer) event.target, event);
+            }
+    }
+
+    public Object[][] buildEventItemList(EntityPlayer player) {
+        ArrayList<Object[]> list = new ArrayList<Object[]>();
+
+        ItemStack[] is = player.inventory.armorInventory;
+        for (int armor = 0; armor < is.length; armor++)
+            if (is[armor] != null && is[armor].getItem() instanceof IItemEvent)
+                list.add(new Object[] { is[armor], 9 + armor });
+
+        if (Loader.isModLoaded("Baubles")) {
```

Comment:
This check should be unnecessary

Reply:
Good point, since baubles is always necessary.